src/librbd/image/RemoveRequest.cc

```diff
@@ -575,6 +577,11 @@ Context *RemoveRequest<I>::handle_mirror_image_remove(int *result) {
     *result = 0;
   }
 
+  if (m_from_trash_purge) {
+    // image was already removed from the directory
```

Comment:
Not sure if I'm following. If `m_from_trash_purge` is true, then it means that the image was already removed from the directory, and the "rbd_id.XYZ` was also removed, when `trash_move` was called for that image.

Am I missing some case here?